### PR TITLE
Fixes borg buckling+searching on harm intent

### LINKED ISSUE #46617: Cyborgs still buckling on harm intent
#46364 seems to not be working for the no buckling aspect of harm intent. Click dragging on help intent will pull the target onto you while harm will do as it previously did before the change.

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/46617.

As it turns out, there were TWO places in the code where cyborgs were attempting to buckle people to themselves upon a click+drag, and one of them didn't have the "not a cyborg that's on harm intent" check. I made one of them not try to buckle things to cyborgs (the other one doesn't get called for non-cyborgs), which I think has fixed the problem.

Also, this PR should now make MouseDrop_T return TRUE when it successfully buckles someone to a cyborg, like with most other buckle-able objects. It also makes the can_buckle variable for cyborgs actually do something if it's FALSE (I don't think that it didn't anything if it was FALSE instead of TRUE before).

## Why It's Good For The Game

Bug bad. Also, this makes https://github.com/tgstation/tgstation/pull/46364 fully do what it was intended to do.

## Changelog
:cl: ATHATH
fix: Click+dragging someone onto yourself while on harm intent as a cyborg should now only search them, not buckle them to yourself (for realsies this time).
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Hold on, I think I can make this look a bit cleaner/less like shitcode.

**Comment by ATH1909** [comment]
There, done!

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -22,9 +22,10 @@
 
 /atom/movable/MouseDrop_T(mob/living/M, mob/living/user)
 	. = ..()
-	if(can_buckle && istype(M) && istype(user))
-		if(user_buckle_mob(M, user))
-			return 1
+	if(!iscyborg(src)) //the code for buckling things to cyborgs is inside of robot.dmm, since cyborgs have different conditions for buckling things than most other buckle-able things do
+		if(can_buckle && istype(M) && istype(user))
+			if(user_buckle_mob(M, user))
+				return 1
```
Comment (Qustinnus) [inline code/game/objects/buckling.dm]: return true you noob

Comment (ATH1909) [reply code/game/objects/buckling.dm]: 1 is what this proc originally returned, but okay

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -22,9 +22,10 @@
 
 /atom/movable/MouseDrop_T(mob/living/M, mob/living/user)
 	. = ..()
-	if(can_buckle && istype(M) && istype(user))
-		if(user_buckle_mob(M, user))
-			return 1
+	if(!iscyborg(src)) //the code for buckling things to cyborgs is inside of robot.dmm, since cyborgs have different conditions for buckling things than most other buckle-able things do
```
Comment (nemvar) [inline code/game/objects/buckling.dm]: istype(src) :nauseated_face:

Comment (ATH1909) [reply code/game/objects/buckling.dm]: @nemvar ?

File: code/game/objects/buckling.dm (hunk 1 above)
Comment (Qustinnus) [reply code/game/objects/buckling.dm]: and thats why we have people like u to fix it

**Comment by Surrealaser** [comment]
Yes please, this is so fucking annoying; I currently can't strip people as medborg who are sealed against my hypo or surgery.

**Comment by Difrit** [comment]
Hot take: Borgs shouldn't be able to strip because they don't have hands

File: code/game/objects/buckling.dm (hunk 2 above)
Comment (ATH1909) [reply code/game/objects/buckling.dm]: @nemvar the check here is "_**!**_iscyborg(src)", not "iscyborg(src)"
please let this PR go through this is really annoying for a lot of borgs reeeeee

Comment (AnturK) [reply code/game/objects/buckling.dm]: Cmon, it's not that much effort to add a proc here and override for borgs.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/buckling.dm (hunk 2 above)
Comment (ATH1909) [reply code/game/objects/buckling.dm]: @AnturK Done (I think)

**Comment by ATH1909** [comment]
Wait, cyborgs aren't subtypes of moveable, !@#$. How do I fix this?

**Pull request merged**